text/0000-refactor-sbom.md

```diff
@@ -0,0 +1,496 @@
+
+This RFC proposes optionally relocating SBOM files for buildpack-provided dependencies from the application image to in-toto attestations
+in an OCI registry
+via a new optional lifecycle phase called `sign`.
+
+# Definitions
+[definitions]: #definitions
+
+**SBOM files for buildpack-provided dependencies** - described in detail in [RFC 0095](https://github.com/buildpacks/rfcs/blob/main/text/0095-sbom.md), 
+buildpacks may output SBOM files in known formats at known locations,
+and the lifecycle includes these as a (single) layer in the application image.
+The SBOM files may be image-scoped or layer-scoped.
+The layer-scoped files are restored on future builds so that buildpacks can use them during the build phase.
+
+**image signature** - allows for cryptographically verifying the digest of an OCI image
+
+**in-toto [attestations](https://github.com/in-toto/attestation/blob/main/spec/README.md)** - signed statements about software artifacts
+
+**[cosign](https://github.com/sigstore/cosign/tree/main#cosign)** - tooling for generating image signatures and attestations
+
+**OCI v1.1** - pending new releases to the OCI image and distribution specs, as described [here](https://opencontainers.org/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1/)
+
+Notably, these new specs codify ways of "attaching" arbitrary OCI artifacts to an image -
+ - via the Referrers API if supported by the registry
+   - the manifest for each artifact should contain:
+     - `subject` to point to the image described by the artifact
+ - via Referrers Tag Schema (fallback) if Referrers API is not supported by the registry
+   - a manifest "list" published to tag `sha256-<image digest>` where each list element should contain:
+     - `artifactType` corresponding to the `config.mediaType` in the manifest
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this? Our current approach has a few drawbacks, namely:
+  * It can make application images quite large
+  * It can be hard to find SBOMs for buildpacks-built images; ecosystem tooling such as `cosign download sbom` won't work (see [PR 278](https://github.com/buildpacks/rfcs/pull/278))
+  * There is currently no clear way to associate SBOMs for build and run base images with an application image (today this is unspec'd and entirely up to the platform to manage)
+  
+- What use cases does it support?
+  * As a buildpacks user, I want **signed** SBOMs for my images in an easily consumable format.
+  * ...
+
+- What is the expected outcome?
+  * Smaller application images
+  * Integration with ecosystem tooling
+  * (Eventually) A more complete SBOM for buildpacks-built images
+
+# What it is
+[what-it-is]: #what-it-is
+
+See [How it Works](#how-it-works).
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Setup
+
+The lifecycle will ship with a new `signer` binary.
+In the past we've talked about making this binary totally separate from the lifecycle,
```

Comment:
From 10/5 Working Group - cosign is nevertheless a significant dependency to take into the lifecycle. If cosign took a configuration file (https://github.com/sigstore/cosign/issues/1288 or similar) maybe we could just set that configuration up according to the [mapping](#mapping-layers-to-attestations) defined below. That would take care of the signing part.

For retrieving SBOMs from the previous image, we could:
* explore making verifying attestations the responsibility of the "preparer" binary, if we ever have one; the preparer could write the predicate data as json for the analyzer to find
* put this in the analyzer and hope the dependency footprint would be smaller if we're only verifying signatures (not creating them)